### DISCUSSION TITLE
GUIDE: How to install on Steam Deck

### DISCUSSION THREAD

**Discussion by Moshugan** [sultim-t/xash-rt#157] [Show and tell]
This is a guide on how to install "Half-Life 1: Ray Traced" on the Steam Deck. The game performs surprisingly well with FSR 2.0 Ultra Performance! It can run at a locked smooth 30 fps, and many scenarios can maintain 40 fps as well.

**- First in Gaming Mode:**

1) Install the latest SteamOS version (3.4.6 as of now), which enables raytracing in the first place

2) Install Half-Life (I installed the Steam version)

**- The next steps are in Desktop mode:**

3) Download the files and follow the installation instructions for the RT version found here in Sultim-T's GitHub page

4) Install the flatpak of Lutris from Discovery Software Center 

**- The next steps are within Lutris:**

5) Open Lutris and press "add game" (the big plus in the upper-left corner)

6) Press "locally installed game"

7) In the "Game info" tab give it a name (something like "Half-Life RT" or whatever) and choose "Wine" from the "Runner" options

8a) In the "Game options" tab, in "Executable" option browse to the folder where the "xash3d.exe" is located in (if you istalled the Steam version of HL to the internal drive and installed the mod correctly, then the path should be " /home/deck/.local/share/Steam/steamapps/common/Half-Life/rt_bin/xash3d.exe ")

8b) If you have a hard time finding the folder, then in the browser window click the "options" button in the upper-right corner and check the "show hidden files" option

8c) In the same "Game options" tab you may write "-width 1280 -height 800 +sleeptime 0" in the "Arguments" option (I'm not exactly sure if this is necessary. I also tried to write a lower resolution there, but it seems that full screen resolution will always be the native res)

8d) In the same "Game options" tab choose "64-bit" for the "Prefix Architecture" option

9) No changes needed in the "Runner options" tab

10) In the "System options" tab scroll down to "Environment variables" and then add a new entry there. Write "RADV_PERFTEST" in the "Key" slot, and "rt" in the "Value" slot

11) Click the blue "SAVE" button in the upper-right corner of the window

12a) Now you should have the game appear in the Lutris main window. Select it and check the lower edge of the window. Next to the "Play" button and its up arrow there should be a little icon that represents the "Wine" options. Click its arrow and open the "Winetricks" option

12b) It might take a little while before a window pops up (or appear in the toolbar down below). In the window choose "select the default wineprefix" and press "OK"

12c) In the next window choose "Install a Windows DLL or component" and press "OK"

12d) In the list that opens make sure that "d3dcompiler_43" and "d3dx9" are selected and the also select "dxvk" and "vcrun2019". Then press "OK"

13) If you want to add the game to Steam Deck's Gaming Mode, then click the arrow next to the "play" button and select "create steam shortcut"

14) Now you should be ready to play the game! Either run it from Lutris in Desktop mode, or from Gaming Mode!

I hope this guide is clear and sufficient! :)

**Comment by Moshugan** [comment]
Oh, and if you want to get rid of the new crowbar and return to the original, just delete the file "v_crowbar.mdl" from the folder "rt/valve/models".

**Comment by stephen-cusi** [comment]
`thx`
